sys-boot/grub/grub-2.06-r2.ebuild

```diff
@@ -1,42 +1,84 @@
+
+if [[ ${PV} == 9999  ]]; then
+	GRUB_BOOTSTRAP=1
+fi
 
-CROS_WORKON_PROJECT="flatcar-linux/grub"
-CROS_WORKON_REPO="https://github.com"
-GRUB_AUTOGEN=1  # We start from Git, so always autogen.
+GRUB_AUTOGEN=1
+GRUB_AUTORECONF=1
+PYTHON_COMPAT=( python3_{8..11} )
+WANT_LIBTOOL=none
 
-if [[ ${PV} == 9999 ]]; then
-	KEYWORDS="~amd64 ~arm64 ~x86"
-else
-	CROS_WORKON_COMMIT="ba40c9d246602646b22b0d190f260f2d98c0351b"  # flatcar-master
-	KEYWORDS="amd64 arm64 x86"
+if [[ -n ${GRUB_AUTOGEN} || -n ${GRUB_BOOTSTRAP} ]]; then
+	inherit python-any-r1
+fi
+
+if [[ -n ${GRUB_AUTORECONF} ]]; then
+	inherit autotools
 fi
-inherit cros-workon
 
-if [[ -n ${GRUB_AUTOGEN} ]]; then
-	PYTHON_COMPAT=( python3_{6..10} )
-	WANT_LIBTOOL=none
-	inherit autotools python-any-r1
+inherit bash-completion-r1 flag-o-matic multibuild optfeature toolchain-funcs
+
+if [[ ${PV} != 9999 ]]; then
+	if [[ ${PV} == *_alpha* || ${PV} == *_beta* || ${PV} == *_rc* ]]; then
+		# The quote style is to work with <=bash-4.2 and >=bash-4.3 #503860
+		MY_P=${P/_/'~'}
+		SRC_URI="https://alpha.gnu.org/gnu/${PN}/${MY_P}.tar.xz"
+		S=${WORKDIR}/${MY_P}
+	else
+		SRC_URI="mirror://gnu/${PN}/${P}.tar.xz"
+		S=${WORKDIR}/${P%_*}
+	fi
+	KEYWORDS="amd64 ~arm arm64 ~ia64 ppc ~ppc64 ~riscv sparc x86"
+else
+	inherit git-r3
+	EGIT_REPO_URI="https://git.savannah.gnu.org/git/grub.git"
 fi
 
-inherit autotools bash-completion-r1 flag-o-matic multibuild pax-utils toolchain-funcs versionator
+SRC_URI+=" https://dev.gentoo.org/~floppym/dist/grub-2.06-backports-r1.tar.xz"
+
+PATCHES=(
+	"${WORKDIR}/${P}-backports"
+	"${FILESDIR}"/gfxpayload.patch
+	"${FILESDIR}"/grub-2.02_beta2-KERNEL_GLOBS.patch
+	"${FILESDIR}"/grub-2.06-test-words.patch
+	"${FILESDIR}"/grub-2.06-add-verity-hash.patch
+	"${FILESDIR}"/grub-2.06-add-gpt-partition-scheme.patch
+)
+
 DEJAVU=dejavu-sans-ttf-2.37
-UNIFONT=unifont-9.0.06
+UNIFONT=unifont-12.1.02
 SRC_URI+=" fonts? ( mirror://gnu/unifont/${UNIFONT}/${UNIFONT}.pcf.gz )
 	themes? ( mirror://sourceforge/dejavu/${DEJAVU}.zip )"
 
 DESCRIPTION="GNU GRUB boot loader"
 HOMEPAGE="https://www.gnu.org/software/grub/"
 
 # Includes licenses for dejavu and unifont
-LICENSE="GPL-3 fonts? ( GPL-2-with-font-exception ) themes? ( BitstreamVera )"
+LICENSE="GPL-3+ BSD MIT fonts? ( GPL-2-with-font-exception ) themes? ( CC-BY-SA-3.0 BitstreamVera )"
 SLOT="2/${PVR}"
-IUSE="debug device-mapper doc efiemu +fonts mount multislot nls static sdl test +themes truetype libzfs"
+IUSE="device-mapper doc efiemu +fonts mount nls sdl test +themes truetype libzfs"
+
+GRUB_ALL_PLATFORMS=( coreboot efi-32 efi-64 emu ieee1275 loongson multiboot qemu qemu-mips pc uboot xen xen-32 xen-pvh )
 
-GRUB_ALL_PLATFORMS=( coreboot efi-32 efi-64 emu ieee1275 loongson multiboot qemu qemu-mips pc uboot xen xen-32 )
-GRUB_ALL_PLATFORMS+=( arm64 )  # Tack on arm64 for pulling in the cross-compiler.
+# Flatcar: Add arm64 to the list of platforms
+GRUB_ALL_PLATFORMS+=( arm64 )
```

Comment:
Do we still need this modification? My guess would be that it comes from times when grub in gentoo was not supported. I also don't understand why this was used for pulling in a cross-compiler…